### ISSUE TITLE
VER: NAV/NAZ 1.7.3 Beeper does not work

### ISSUE THREAD

**Issue by itswally1** [iNavFlight/inav#2864]
The beeper will not work. In the mode selection the beeper is enabled and when the proper switch is positioned it lines up under the beeper selection.
The board is EMAX SKYLINE32
The CLI says beeper disabled:None
Also on the CLI if i type
 beeper -on
The response is:
Disabled ON_USB
The beeper is assighed channel 6 and the output on the beeper pins goes from .9 volts to .3 volts

How can I enable the beeper?

Thanks in advance for your support

**Comment by shellixyz** [comment]
You don't need to enable the beeper it is enabled by default. It should sound when the `BEEPER` flight mode is enabled. You should should check your wiring and beeper polarity and health. If the wiring is good and your beeper is working when 5V is applied to it your board is probably defective.

**Comment by itswally1** [comment]
Thanks for the quick response.
The board/beeper works with Cleanflight not with INAV.
INAV has better stability so that is why I am using it. 
In my note I have read the voltage at the beeper and beeper unit works with 5v applied.
The flight mode has been set properly

So why when I type beeper and the flight mode is on for the beeper
does the screen respond with:
 beeper disabled:None
and
Disabled ON_USB
???????
Regards,
Don

**Comment by shellixyz** [comment]
Disabled ON_USB probably means it is disabled when you connect the board with an USB cable.

Does your board beep when you plug the battery ? It should beep at startup. Fast beeps then 3 slow beeps after the run-time calibration.
What target are you using with Cleanflight ?

Please post a dump of your iNav settings.

**Comment by itswally1** [comment]
No beeps. Currently using INAV
Here is the dump
Thanks in advance for looking at the dump
============================
# version
# INAV/NAZE 1.7.3 Sep  9 2017 / 09:59:03 (974440df)

# dump
# INAV/NAZE 1.7.3 Sep  9 2017 / 09:59:03 (974440df)
mixer QUADX

mmix reset

servo 0 1000 2000 1500 100 -1 
servo 1 1000 2000 1500 100 -1 
servo 2 1000 2000 1500 100 -1 
servo 3 1000 2000 1500 100 -1 
servo 4 1000 2000 1500 100 -1 
servo 5 1000 2000 1500 100 -1 
servo 6 1000 2000 1500 100 -1 
servo 7 1000 2000 1500 100 -1 
smix reset

feature -RX_PPM
feature -VBAT
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -TELEMETRY
feature -CURRENT_METER
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DASHBOARD
feature -BLACKBOX
feature -CHANNEL_FORWARDING
feature -TRANSPONDER
feature -AIRMODE
feature -SUPEREXPO
feature -VTX
feature -RX_SPI
feature -SOFTSPI
feature -PWM_SERVO_DRIVER
feature -PWM_OUTPUT_ENABLE
feature -OSD
feature RX_PPM
feature PWM_OUTPUT_ENABLE
beeper RUNTIME_CALIBRATION
beeper HW_FAILURE
beeper RX_LOST
beeper RX_LOST_LANDING
beeper DISARMING
beeper ARMING
beeper ARMING_GPS_FIX
beeper BAT_CRIT_LOW
beeper BAT_LOW
beeper GPS_STATUS
beeper RX_SET
beeper ACTION_SUCCESS
beeper ACTION_FAIL
beeper READY_BEEP
beeper MULTI_BEEPS
beeper DISARM_REPEAT
beeper ARMED
beeper SYSTEM_INIT
beeper ON_USB
beeper LAUNCH_MODE
map AETR1234
name -
serial 0 1 115200 38400 0 115200
serial 1 0 115200 38400 0 115200
aux 0 0 0 1900 2100
aux 1 1 1 900 1350
aux 2 2 1 1300 1700
aux 3 11 1 1800 2100
aux 4 0 0 900 900
aux 5 0 0 900 900
aux 6 0 0 900 900
aux 7 0 0 900 900
aux 8 0 0 900 900
aux 9 0 0 900 900
aux 10 0 0 900 900
aux 11 0 0 900 900
aux 12 0 0 900 900
aux 13 0 0 900 900
aux 14 0 0 900 900
aux 15 0 0 900 900
aux 16 0 0 900 900
aux 17 0 0 900 900
aux 18 0 0 900 900
aux 19 0 0 900 900
adjrange 0 0 0 900 900 0 0
adjrange 1 0 0 900 900 0 0
adjrange 2 0 0 900 900 0 0
adjrange 3 0 0 900 900 0 0
adjrange 4 0 0 900 900 0 0
adjrange 5 0 0 900 900 0 0
adjrange 6 0 0 900 900 0 0
adjrange 7 0 0 900 900 0 0
adjrange 8 0 0 900 900 0 0
adjrange 9 0 0 900 900 0 0
adjrange 10 0 0 900 900 0 0
adjrange 11 0 0 900 900 0 0
rxrange 0 1000 2000
rxrange 1 1000 2000
rxrange 2 1000 2000
rxrange 3 1000 2000
set looptime = 2000
set gyro_sync = OFF
set gyro_sync_denom = 2
set align_gyro = DEFAULT
set gyro_hardware_lpf = 42HZ
set gyro_lpf_hz = 60
set moron_threshold = 32
set vbat_adc_channel = 2
set rssi_adc_channel = 3
set current_adc_channel = 1
set airspeed_adc_channel = 0
set align_acc = DEFAULT
set acc_hardware = MPU6050
set acc_lpf_hz = 15
set acczero_x = 151
set acczero_y = -18
set acczero_z = -164
set accgain_x = 4063
set accgain_y = 4093
set accgain_z = 4045
set align_mag = DEFAULT
set mag_hardware = HMC5883
set mag_declination = 0
set magzero_x = 107
set magzero_y = -112
set magzero_z = -141
set mag_calibration_time = 30
set baro_hardware = MS5611
set baro_use_median_filter = ON
set mid_rc = 1500
set min_check = 1100
set max_check = 1900
set rssi_channel = 0
set rssi_scale = 30
set rssi_invert = OFF
set rc_smoothing = ON
set serialrx_provider = SPEK1024
set sbus_inversion = ON
set rx_spi_rf_channel_count = 0
set rx_min_usec = 885
set rx_max_usec = 2115
set blackbox_rate_num = 1
set blackbox_rate_denom = 1
set blackbox_device = SERIAL
set min_throttle = 1150
set max_throttle = 1850
set min_command = 1000
set motor_pwm_rate = 400
set motor_pwm_protocol = STANDARD
set failsafe_delay = 5
set failsafe_recovery_delay = 5
set failsafe_off_delay = 200
set failsafe_throttle = 1000
set failsafe_throttle_low_delay = 100
set failsafe_procedure = SET-THR
set failsafe_stick_threshold = 50
set failsafe_fw_roll_angle = -200
set failsafe_fw_pitch_angle = 100
set failsafe_fw_yaw_rate = -45
set align_board_roll = 0
set align_board_pitch = 0
set align_board_yaw = 0
set gimbal_mode = NORMAL
set battery_capacity = 0
set vbat_scale = 10
set vbat_max_cell_voltage = 43
set vbat_min_cell_voltage = 33
set vbat_warning_cell_voltage = 35
set current_meter_scale = 400
set current_meter_offset = 0
set multiwii_current_meter_output = OFF
set current_meter_type = ADC
set yaw_motor_direction = 1
set yaw_jump_prevention_limit = 200
set 3d_deadband_low = 1406
set 3d_deadband_high = 1514
set 3d_neutral = 1460
set servo_center_pulse = 1500
set servo_pwm_rate = 50
set servo_lpf_hz = 20
set flaperon_throw_offset = 200
set tri_unarmed_servo = ON
set reboot_character = 82
set imu_dcm_kp = 2500
set imu_dcm_ki = 50
set imu_dcm_kp_mag = 10000
set imu_dcm_ki_mag = 0
set small_angle = 25
set fixed_wing_auto_arm = OFF
set disarm_kill_switch = ON
set auto_disarm_delay = 5
set gps_provider = UBLOX
set gps_sbas_mode = NONE
set gps_dyn_model = AIR_1G
set gps_auto_config = ON
set gps_auto_baud = ON
set gps_min_sats = 6
set deadband = 5
set yaw_deadband = 5
set pos_hold_deadband = 20
set alt_hold_deadband = 50
set 3d_deadband_throttle = 50
set fw_autotune_overshoot_time = 100
set fw_autotune_undershoot_time = 200
set fw_autotune_threshold = 50
set fw_autotune_ff_to_p_gain = 10
set fw_autotune_ff_to_i_tc = 600
set inav_gravity_cal_tolerance = 5
set inav_use_gps_velned = ON
set inav_gps_delay = 200
set inav_reset_altitude = FIRST_ARM
set inav_max_surface_altitude = 200
set inav_w_z_surface_p =  3.500
set inav_w_z_surface_v =  6.100
set inav_w_z_baro_p =  0.350
set inav_w_z_gps_p =  0.200
set inav_w_z_gps_v =  0.500
set inav_w_xy_gps_p =  1.000
set inav_w_xy_gps_v =  2.000
set inav_w_z_res_v =  0.500
set inav_w_xy_res_v =  0.500
set inav_w_acc_bias =  0.010
set inav_max_eph_epv =  1000.000
set inav_baro_epv =  100.000
set nav_disarm_on_landing = OFF
set nav_use_midthr_for_althold = OFF
set nav_extra_arming_safety = ON
set nav_user_control_mode = ATTI
set nav_position_timeout = 5
set nav_wp_radius = 100
set nav_wp_safe_distance = 10000
set nav_auto_speed = 300
set nav_auto_climb_rate = 500
set nav_manual_speed = 500
set nav_manual_climb_rate = 200
set nav_landing_speed = 200
set nav_land_slowdown_minalt = 500
set nav_land_slowdown_maxalt = 2000
set nav_emerg_landing_speed = 500
set nav_min_rth_distance = 500
set nav_rth_climb_first = ON
set nav_rth_climb_ignore_emerg = OFF
set nav_rth_tail_first = OFF
set nav_rth_allow_landing = ON
set nav_rth_alt_mode = AT_LEAST
set nav_rth_abort_threshold = 50000
set nav_rth_altitude = 1000
set nav_mc_bank_angle = 30
set nav_mc_hover_thr = 1500
set nav_mc_auto_disarm_delay = 2000
set nav_fw_cruise_thr = 1400
set nav_fw_min_thr = 1200
set nav_fw_max_thr = 1700
set nav_fw_bank_angle = 20
set nav_fw_climb_angle = 20
set nav_fw_dive_angle = 15
set nav_fw_pitch2thr = 10
set nav_fw_loiter_radius = 5000
set nav_fw_launch_velocity = 300
set nav_fw_launch_accel = 1863
set nav_fw_launch_max_angle = 45
set nav_fw_launch_detect_time = 40
set nav_fw_launch_thr = 1700
set nav_fw_launch_idle_thr = 1000
set nav_fw_launch_motor_delay = 500
set nav_fw_launch_spinup_time = 100
set nav_fw_launch_timeout = 5000
set nav_fw_launch_climb_angle = 18
set telemetry_switch = OFF
set telemetry_inversion = OFF
set frsky_default_latitude =  0.000
set frsky_default_longitude =  0.000
set frsky_coordinates_format = 0
set frsky_unit = METRIC
set frsky_vfas_precision = 0
set frsky_vfas_cell_voltage = OFF
set hott_alarm_sound_interval = 5
set ibus_telemetry_type = 0
set ltm_update_rate = NORMAL
set i2c_speed = 400KHZ
set cpu_underclock = OFF
set debug_mode = NONE
set throttle_tilt_comp_str = 0
set input_filtering_mode = OFF
set mode_range_logic_operator = OR
profile 1

set mc_p_pitch = 40
set mc_i_pitch = 30
set mc_d_pitch = 23
set mc_p_roll = 40
set mc_i_roll = 30
set mc_d_roll = 23
set mc_p_yaw = 85
set mc_i_yaw = 45
set mc_d_yaw = 0
set mc_p_level = 20
set mc_i_level = 15
set mc_d_level = 75
set fw_p_pitch = 5
set fw_i_pitch = 7
set fw_ff_pitch = 50
set fw_p_roll = 5
set fw_i_roll = 7
set fw_ff_roll = 50
set fw_p_yaw = 6
set fw_i_yaw = 10
set fw_ff_yaw = 60
set fw_p_level = 20
set fw_i_level = 5
set fw_d_level = 75
set max_angle_inclination_rll = 300
set max_angle_inclination_pit = 300
set dterm_lpf_hz = 40
set yaw_lpf_hz = 30
set dterm_setpoint_weight =  0.000
set fw_iterm_throw_limit = 165
set fw_reference_airspeed =  1000.000
set fw_turn_assist_yaw_gain =  1.000
set pidsum_limit = 500
set yaw_p_limit = 300
set iterm_ignore_threshold = 200
set yaw_iterm_ignore_threshold = 50
set rate_accel_limit_roll_pitch = 0
set rate_accel_limit_yaw = 10000
set heading_hold_rate_limit = 90
set nav_mc_pos_z_p = 50
set nav_mc_pos_z_i = 0
set nav_mc_pos_z_d = 0
set nav_mc_vel_z_p = 100
set nav_mc_vel_z_i = 50
set nav_mc_vel_z_d = 10
set nav_mc_pos_xy_p = 65
set nav_mc_pos_xy_i = 120
set nav_mc_pos_xy_d = 10
set nav_mc_vel_xy_p = 180
set nav_mc_vel_xy_i = 15
set nav_mc_vel_xy_d = 100
set nav_fw_pos_z_p = 50
set nav_fw_pos_z_i = 0
set nav_fw_pos_z_d = 0
set nav_fw_pos_xy_p = 75
set nav_fw_pos_xy_i = 5
set nav_fw_pos_xy_d = 8
set rc_expo = 70
set rc_yaw_expo = 20
set thr_mid = 50
set thr_expo = 0
set roll_rate = 20
set pitch_rate = 20
set yaw_rate = 20
set tpa_rate = 0
set tpa_breakpoint = 1500

**Comment by shellixyz** [comment]
I can't try to reproduce since I don't have a Naze FC anymore. All I know is that it was working when I was using one. Sorry but the boards using F1 MCU like the Naze are not supported anymore you should consider upgrading to a F3, F4 or F7 board.

**Comment by stronnag** [comment]
works fine on my bench test naze32 (Flip32 h/w) with 1.7.3. 

But as shellixyz wrote, this is obsolete firmware on an unsupported FC.

**Comment by SPlissken** [comment]
I got the same problem moving from 1.7.2 to 1.7.3:
Beeper not sounding on NAZE 1.7.3 #2252

I had a poke around in the code but I couldn't immediately see the issue.

**Comment by shellixyz** [comment]
@SPlissken It's odd nothing changed between 1.7.2 and 1.7.3 related to beeper handling in the general code or the NAZE target.

**Comment by SPlissken** [comment]
@shellixyz definitely odd, I went a little crazy chasing it at the time: ghosts in the interrupts.

It was really easy to reproduce on the bench: erase and flash naked Naze to 1.7.2 beeping, erase and flash naked Naze to 1.7.3 no beeping.

By naked I mean just USB, 5V and beeper connected.

**Comment by itswally1** [comment]
I changed my four quads back to 1,7,2 and the beeper works. So the issue is with 1.7.3 code keeping the beeper from working.

Thanks all for you support and hopefully the issue with 1.7.3 will be fixed in the future

**Comment by DzikuVx** [comment]
Unfortunately we are no longer supporting NAZE32 or any other F1 board. That mean you probably will have to stick with 1.7.2 or upgrade flight controller to F3/F4/F7 family

**Comment by VincenzoScrima** [comment]
Hello, I know that this issue is closed but I thought I could help someone with same problem.
I own a Naze32 board and I had the beeper not working with INAV 1.7.3 (974440d).
I solved it rebuilding the 1.7.3 version after having removed all the code related to the cpu_underclock.  This code is introduced after the version with commit no. 782b482 on Sept. 8 2018.
An easy way to do it is to copy the follwing files from the 782b482 version and replace them with the 974440d ones, recompile it and that's it! The beeper should be working again.

Theese are the files:

src/main/drivers/system.h
src/main/drivers/system_stm32f10x.c
src/main/fc/config.c
src/main/fc/config.h
src/main/fc/fc_init.c
src/main/fc/settings.yaml
src/main/target/common.h
src/main/vcp/hw_config.c

The underclock feature is only for F3 and F4 processor anyway.
I did this way just because I wanted to keep all the other changes.
Kindly let me know if I did something bad or if underclocking is useful for the Naze32. 
Greetings! 
:-)
